src/components/AttachmentModal.js

```diff
@@ -42,28 +45,91 @@ const propTypes = {
 };
 
 const defaultProps = {
-    title: '',
-    sourceURL: null,
-    onConfirm: null,
-    isAuthTokenRequired: false,
+    sortedReportActions: [],
+    visibleAttachment: {
+        sourceURL: null,
+        isAttachment: false,
+        isModalOpen: false,
+        file: {name: ''},
+    },
 };
 
 class AttachmentModal extends Component {
     constructor(props) {
         super(props);
 
+        this.handleKeyPress = this.handleKeyPress.bind(this);
+        this.getNextAttachment = this.getNextAttachment.bind(this);
+        this.getAttachmentSource = this.getAttachmentSource.bind(this);
+
         this.state = {
-            isModalOpen: false,
-            file: null,
-            sourceURL: props.sourceURL,
+            title: 'Attachment',
         };
     }
 
+    componentDidUpdate(prevProps) {
+        const {visibleAttachment: {isModalOpen}} = this.props;
+        if (!prevProps.visibleAttachment.isModalOpen && isModalOpen) {
+            document.addEventListener('keydown', this.handleKeyPress, false);
+        } else {
+            document.removeEventListener('keydown', this.handleKeyPress, false);
+        }
+    }
+
+    // Using regex to get the src attribute from the html img tag
+    getAttachmentSource(htmlString) {
+        const imgRegx = /<img.*?data-expensify-source="(.*?)"[^>]+>/g;
+        const imageGroup = imgRegx.exec(htmlString);
+        return imageGroup[1];
+    }
+
+    // A function to get the next attachment based on the current attachment being show in the report.
+    getNextAttachment = (toRight) => {
```

Comment:
NAB, I think this would be clearer if we used a constant to indicate which direction we are moving here and then pass `'right'` or `'left'`.

Reply:
But if we're not moving to the right, then the assumption is we're moving to the left.

Reply:
ok